lib/davinci_crd_test_kit/client_tests/hook_request_valid_context_test.rb

```diff
@@ -49,15 +49,25 @@ def hook_name
     end
 
     run do
-      assert_valid_json(request.request_body)
-      request_body = JSON.parse(request.request_body)
+      assert_valid_json(contexts_prefetches)
+      hook_contexts_prefetches = JSON.parse(contexts_prefetches)
+      assert(hook_contexts_prefetches.any? { |context_prefetch| context_prefetch['context'].present? },
+             "No #{hook_name} requests contained the `context` field.")
+      error_messages = []
+      hook_contexts_prefetches.each_with_index do |context_prefetch, index|
+        context = context_prefetch['context']
+        assert(context.present?, 'Missing required context field.')
+        assert(context.is_a?(Hash), 'Context is in an incorrect format.')
+        hook_request_context_check(context, hook_name, index + 1)
+        no_error_validation('Context is not valid.')
+      rescue Inferno::Exceptions::AssertionException => e
+        error_messages << "Request #{index + 1}: #{e.message}"
+      end
```

Comment:
I think this test should skip if there are no contexts to validate. Also I don't see the need to validate the context data type here again as it was already done in the required field validation. 

Also `no_error_validation` should be called at the end. no need to rescue assertion error and maintain another list of error messages.

Reply:
I agree with all of this and made these changes, the only thing I did not remove was validating the context data type here again even though you are right, it was already done in the required field validation. The reason I have this here though is to ensue the test does not throw an error if a context is in an incorrect format

Reply:
There's no need because when you're collecting the context in the required field test, you only do so if it's a hash. So the contexts output can either be an empty array or array of hashes.

Reply:
Right, I was thinking about how it used to work, but you are right now that the context is an output of the required fields test, this is no longer needed